### ISSUE TITLE
PHP Parse error in snippet List

### ISSUE THREAD

**Issue by microcipcip** [ClipperCMS/ClipperCMS#413]
I have installed ClipperCMS Develop. I am getting this issue " PHP Parse error in snippet List" with the demo website.

```
ClipperCMS encountered the following error while attempting to parse the requested resource:
« Fatal error: syntax error, unexpected '{' »

PHP error debug
  Error:    syntax error, unexpected '{'     
  Error type/ Nr.:  Parsing Error - 4    
  File:     C:\xampp\htdocs\ClipperCMS-develop\manager\includes\document.parser.class.inc.php(1294) : eval()'d code  
  Line:     172
```

**Comment by microcipcip** [comment]
I get the error only where there is List running. The other pages works all right.

**Comment by TimGS** [comment]
Puzzled. Try pasting in install/assets/snippets/list.tpl into the List snippet via the manager, or just reinstall list via the installer upgrade option.

**Comment by microcipcip** [comment]
I have tried to install List from list.tpl and I have also installed ClipperCMS Develop from scratch other two times, changing database type. MyISAM and Inno db. I am still getting the error. Is there anything I can do to report a more detailed error?

**Comment by microcipcip** [comment]
My PHP Version 5.4.7
Database Version 5.5.27

**Comment by TimGS** [comment]
If you paste in list.tpl, which line number is reported for the error?

**Comment by microcipcip** [comment]
You mean if I paste the code from list.tpl to the snippet in the manager? The same line.

**Comment by TimGS** [comment]
Have you pasted the entire file including comments?

Line 172 of list.tpl doesn't have any {curly braces}

**Comment by microcipcip** [comment]
Yes, I have pasted the whole file, I have just deleted the first comments right before the <php? tag

**Comment by TimGS** [comment]
Do you get the same error if you move the site to a PHP 5.3 server?

**Comment by Cipa** [comment]
just a thought. can you add a var_dump before line 1293
var_dump($___snippet_code);

**Comment by microcipcip** [comment]
Done, I have got a huge code, should I paste it all here?

**Comment by Cipa** [comment]
Well I am not sure but I think that document.parser.class.inc.php(1294)
means there is an error with the eval on that line. Maybe we can find out
from that code where is the extra { that the error is referring to.
Maybe see line 172 of that code if it makes sense

**Comment by microcipcip** [comment]
Line 172 is just a comment. Is List working all right in your develop install?

**Comment by Cipa** [comment]
I don't have a working install right now.

**Comment by microcipcip** [comment]
Do you have any other suggestion?

**Comment by Cipa** [comment]
Not really, I just setup a new install and there I have a problem as well

Basically pages that use Ditto or List don't work. A simple test would be
to just type [[Ditto]] in the rich textarea of the home page of a newly
installed ClipperCMS. I also think other pages using Ditto don't work.

Error is

PHP error debug
  Error: Cannot modify header information - headers already sent by (output
started at
/Users/Dropbox/Sites/ClipperCMS/manager/includes/document.parser.class.inc.php:3987)

  Error type/ Nr.: Warning - 2
  File: DocumentParser in snippet List
  Line: 3981

I'm on php 5.5.3

**Comment by Cipa** [comment]
I can now confirm I get the same error and it is Ditto/List. Don't know why though

**Comment by Cipa** [comment]
The problem is on line 172 of the List snippet(the code in the manager) and it is related to the lang files. If I remove the for loop at that line everything starts to work. Maybe one of the latest lang changes broke List. Will have a look later this week unless a fix is found by then.

Cheers

**Comment by Cipa** [comment]
Note: There's an extra ( on the if but that doesn't fix the problem.

**Comment by TimGS** [comment]
Does this exist on branch 1.2.x?

**Comment by Cipa** [comment]
I downloaded the latest develop I think. it says _1.2.6 Leander_

**Comment by TimGS** [comment]
Does it exist on branch clipper-1.2.x as on github?

**Comment by microcipcip** [comment]
I'll try now

**Comment by microcipcip** [comment]
Yes, the same error. I have just downloaded the develop version.

**Comment by TimGS** [comment]
Some language files have been updated recently in clipper-1.2.x. Can you recheck?

Can you also check whether the behaviour is the same for different languages?

-- Tim.

**Comment by Cipa** [comment]
The problem is with convertLanguageArray function

List passes 
$modx->convertLanguageArray('_lang', $language_files['base_language'], '_lang')

but the function expects

```
 * @param array &$cla_conversion_lang Array to convert
 * @param string $fallback filename of fallback file
 * @param string $fallback_var name of array in fallback file
```

Check the first param - it expects an array. 

So the fix in my opinion is to change line 175
from

```
if (!$modx->convertLanguageArray('_lang', $language_files['base_language'], '_lang')) {
```

to

```
if (!$modx->convertLanguageArray($_lang, $language_files['base_language'], '_lang')) {
```

**Comment by TimGS** [comment]
I think $modx->convertLanguageArray() is on branch develop, but not clipper-1.2.x, so I've re-assigned this issue to 1.3.

Is this right? Is branch clipper-1.2.x unaffected by this bug?

-- Tim.

**Comment by Cipa** [comment]
I think it's only develop

**Comment by kp52** [comment]
Problem is in develop only, List v2.1.3. Two problems, in fact:
- syntax error due to double opening parenthesis around line 172. Reported as unexpected "{" because "if" hasn't seen a closing parenthesis yet.
- as noted by cipa, string instead of variable as first parameter in line 175.
  Demo site works normally with these fixed, as per pull request #421.

**Comment by TimGS** [comment]
Now fixed in develop c/o @kp52
